packages/shared-ux/avatar/user_profile/impl/user_profile.ts

```diff
@@ -0,0 +1,145 @@
+ * `x-pack/plugins/security/common/model/user_profile.ts`
+ *
+ * When making changes please ensure to keep both files in sync.
+ */
+
+/**
+ * Describes basic properties stored in user profile.
+ */
+export interface UserProfile<D extends UserProfileData = UserProfileData> {
+  /**
+   * Unique ID for of the user profile.
+   */
+  uid: string;
+
+  /**
+   * Indicates whether user profile is enabled or not.
+   */
+  enabled: boolean;
+
+  /**
+   * Information about the user that owns profile.
+   */
+  user: UserProfileUserInfo;
+
+  /**
+   * User specific data associated with the profile.
+   */
+  data: Partial<D>;
+}
+
+/**
+ * Basic user information returned in user profile.
+ */
+export interface UserProfileUserInfo {
+  /**
+   * Username of the user.
+   */
+  username: string;
+  /**
+   * Optional email of the user.
+   */
+  email?: string;
+  /**
+   * Optional full name of the user.
+   */
+  full_name?: string;
+  /**
+   * Optional display name of the user.
+   */
+  display_name?: string;
+}
+
+/**
+ * Placeholder for data stored in user profile.
+ */
+export type UserProfileData = Record<string, unknown>;
+
+/**
+ * Avatar stored in user profile.
+ */
+export interface UserProfileAvatarData {
+  /**
+   * Optional initials (two letters) of the user to use as avatar if avatar picture isn't specified.
+   */
+  initials?: string;
+  /**
+   * Background color of the avatar when initials are used.
+   */
+  color?: string;
+  /**
+   * Base64 data URL for the user avatar image.
+   */
+  imageUrl?: string;
+}
+
+export const USER_AVATAR_FALLBACK_CODE_POINT = 97; // code point for lowercase "a"
+export const USER_AVATAR_MAX_INITIALS = 2;
+
+/**
+ * Determines the color for the provided user profile.
+ * If a color is present on the user profile itself, then that is used.
+ * Otherwise, a color is provided from EUI's Visualization Colors based on the display name.
+ *
+ * @param {UserProfileUserInfo} user User info
+ * @param {UserProfileAvatarData} avatar User avatar
+ */
+export function getUserAvatarColor(
+  user: Pick<UserProfileUserInfo, 'username' | 'full_name'>,
+  avatar?: UserProfileAvatarData
+) {
+  if (avatar && avatar.color) {
+    return avatar.color;
+  }
+
+  const firstCodePoint = getUserDisplayName(user).codePointAt(0) || USER_AVATAR_FALLBACK_CODE_POINT;
+
+  return VISUALIZATION_COLORS[firstCodePoint % VISUALIZATION_COLORS.length];
+}
+
+/**
+ * Determines the initials for the provided user profile.
+ * If initials are present on the user profile itself, then that is used.
+ * Otherwise, the initials are calculated based off the words in the display name, with a max length of 2 characters.
+ *
+ * @param {UserProfileUserInfo} user User info
+ * @param {UserProfileAvatarData} avatar User avatar
+ */
+export function getUserAvatarInitials(
+  user: Pick<UserProfileUserInfo, 'username' | 'full_name'>,
+  avatar?: UserProfileAvatarData
+) {
+  if (avatar && avatar.initials) {
+    return avatar.initials;
```

Comment:
nit: can be shortened to `avatar?.initials`

Reply:
[45aa4b9](https://github.com/elastic/kibana/pull/139247/commits/45aa4b98aca28e812d906c57a733886fb8661213) for changes 🙇🏻‍♀️